chromium_src/chrome/browser/profiles/profile.h

```diff
@@ -0,0 +1,15 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_CHROME_BROWSER_PROFILES_PROFILE_H_
+#define BRAVE_CHROMIUM_SRC_CHROME_BROWSER_PROFILES_PROFILE_H_
+
+#include "../../../../../chrome/browser/profiles/profile.h"
+
+// This is meant as a replacement for profile->GetProfileType() ==
+// GUEST_PROFILE, since GetProfileType is no longer public.
+bool IsGuestProfile(Profile* profile);
```

Comment:
I think this is a good place to put this and I'd like to change `Profile::IsTorProfile()` -> `IsTorProfile(Profile* profile)` to get rid of that patch in a followup

Reply:
actually shouldn't we be using `IsGuestSession`?

Reply:
addressed in cf2d8e5629839c23a8f7db0e95b77b857a0ff12a